File: test-schemas-domain-classes/src/main/scala/flatgraph/testdomains/generic/PropertyNames.java
package flatgraph.testdomains.generic;

import java.util.HashSet;
import java.util.Set;

public class PropertyNames {


public static final String int_list = "int_list";


public static final String int_mandatory = "int_mandatory";


public static final String int_optional = "int_optional";


public static final String string_list = "string_list";


public static final String string_mandatory = "string_mandatory";


public static final String string_optional = "string_optional";

public static Set<String> ALL = new HashSet<String>() {{
add(int_list);
add(int_mandatory);
add(int_optional);
add(string_list);
add(string_mandatory);
add(string_optional);
}};

}

File: test-schemas-domain-classes-scala3/src/main/scala/testdomains/generic/PropertyNames.scala
package testdomains.generic

import java.util.{HashSet, Set}
import scala.jdk.CollectionConverters.SeqHasAsJava

object PropertyNames {
  val BooleanOptional: String = "boolean_optional"
  val DoubleOptional: String  = "double_optional"
  val FloatOptional: String   = "float_optional"
  val IntList: String         = "int_list"
  val IntMandatory: String    = "int_mandatory"
  val IntOptional: String     = "int_optional"
  val LongOptional: String    = "long_optional"
  val StringList: String      = "string_list"
  val StringMandatory: String = "string_mandatory"
  val StringOptional: String  = "string_optional"

  /** This is a contained node */
  val ContainedNodeB: String = "contained_node_b"

  val All: Set[String] = new HashSet[String](
    Seq(
      BooleanOptional,
      DoubleOptional,
      FloatOptional,
      IntList,
      IntMandatory,
      IntOptional,
      LongOptional,
      StringList,
      StringMandatory,
      StringOptional,
      ContainedNodeB
    ).asJava
  )
}
